packages/multiselect/src/react/use-keybinding.ts

```diff
@@ -0,0 +1,91 @@
+import { useCallback, useEffect, useMemo, useReducer } from 'react'
+import { isString } from '@pluralsight/ps-design-system-util'
+import { switchcase } from './utils'
+
+type Callback = () => void
+type State = Record<string, boolean>
+type Action = { type: string; data?: State; key?: string }
+
+export function useKeybinding(
+  binding: string[],
+  callback: Callback,
+  el: HTMLElement | Document = document
+) {
+  const initialMapping = useMemo(() => {
+    const map = binding.reduce<Record<string, boolean>>(
+      (acc, key) => ({
+        ...acc,
+        [key.toLowerCase()]: false
+      }),
+      {}
+    )
+
+    return map
+  }, [binding])
+
+  const [keys, setKeys] = useReducer(reducer, initialMapping)
```

Comment:
Why is tracking the pressed keys in internal state required?

Reply:
so i can get a combo. alt+up for example. this was the best approach i could come up with but i'm open to alternatives

Reply:
Could you use just a regular eventHandler that detects the alt key?

Pseudo code:

```
useEffect
  node.addEventListener 'keydown', evt => 
    if evt.key is down arrow and evt.altKey
      open menu
```

Reply:
🤦  yeah...i'll do that. thanks for the coffeescript :)

Reply:
updated!